pandas/core/generic.py

```diff
@@ -6224,12 +6224,18 @@ def astype(
 
             results = []
             for i, (col_name, col) in enumerate(self.items()):
-                cdt = dtype_ser.iat[i]
-                if isna(cdt):
-                    res_col = col.copy() if copy else col
-                else:
-                    res_col = col.astype(dtype=cdt, copy=copy, errors=errors)
-                results.append(res_col)
+                try:
```

Comment:
1. Better if this `try/except` only applies to the line where we know the conversion fails
2. Could the more specific `Exception`s be captured?

Reply:
done